ext/standard/image.c

```diff
@@ -1408,6 +1418,11 @@ PHPAPI int php_getimagetype(php_stream *stream, const char *input, char *filetyp
 		return IMAGE_FILETYPE_IFF;
 	} else if (!memcmp(filetype, php_sig_ico, 4)) {
 		return IMAGE_FILETYPE_ICO;
+	} else if (!memcmp(filetype, php_sig_heifheic, 12) ||
```

Comment:
The comment above indicates that only 4 bytes have been read, and a cursory glance shows me that indeed that is the case. So this memcmp won't do the job, I think you need to read additional bytes.

Reply:
AFAIK, it s far from finished, gd does not support all HEIF formats (e.g. multi image format). imagecreatefromstring overall for this not gonna work for now (I wrote an email on internals).

Reply:
Yeah I went here because I saw your email, I guess I should've waited a bit before reading the code.